tests/i18n/test_extraction.py

```diff
@@ -27,23 +27,36 @@
 
 LOCALE = 'de'
 has_xgettext = find_command('xgettext')
-this_directory = os.path.dirname(upath(__file__))
+source_code_dir = os.path.dirname(upath(__file__))
 
 
-@skipUnless(has_xgettext, 'xgettext is mandatory for extraction tests')
-class ExtractorTests(SerializeMixin, SimpleTestCase):
+class POFileAssertionMixin(object):
 
-    # makemessages scans the current working directory and writes in the
-    # locale subdirectory. There aren't any options to control this. As a
-    # consequence tests can't run in parallel. Since i18n tests run in less
-    # than 4 seconds, serializing them with SerializeMixin is acceptable.
-    lockfile = __file__
+    def _assertPoKeyword(self, keyword, expected_value, haystack, use_quotes=True):
+        q = '"'
+        if use_quotes:
+            expected_value = '"%s"' % expected_value
+            q = "'"
+        needle = '%s %s' % (keyword, expected_value)
+        expected_value = re.escape(expected_value)
+        return self.assertTrue(re.search('^%s %s' % (keyword, expected_value), haystack, re.MULTILINE),
+                               'Could not find %(q)s%(n)s%(q)s in generated PO file' % {'n': needle, 'q': q})
 
-    test_dir = os.path.abspath(os.path.join(this_directory, 'commands'))
+    def assertMsgId(self, msgid, haystack, use_quotes=True):
+        return self._assertPoKeyword('msgid', msgid, haystack, use_quotes=use_quotes)
+
+
+@skipUnless(has_xgettext, 'xgettext is mandatory for extraction tests')
```

Comment:
My only question is if this skip logic is still correctly applied (i.e. none of the other classes that inherited `ExtractorTests` require it)? If you verify that, ship it.